Actions/Sign/Sign.psm1

```diff
@@ -1,44 +1,99 @@
-function GetNavSipFromArtifacts
-(
-    [string] $NavSipDestination
-)
-{
-    $artifactTempFolder = Join-Path $([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
+<#
+    .SYNOPSIS
+    Installs the dotnet signing tool.
+    .DESCRIPTION
+    Installs the dotnet signing tool if isn't already installed.
+    If you want to avoid installing it on-demand while signing you can run ´dotnet tool install sign --version $Version --global´ to install it on your machine
+    .PARAMETER Version
+    The version of the signing tool to install.
+#>
+function Install-SignTool() {
+        param(
+            [Parameter(Mandatory = $false)]
+            [string] $Version
+        )
 
-    try {
-        Download-Artifacts -artifactUrl (Get-BCArtifactUrl -type Sandbox -country core) -basePath $artifactTempFolder | Out-Null
-        Write-Host "Downloaded artifacts to $artifactTempFolder"
-        $navsip = Get-ChildItem -Path $artifactTempFolder -Filter "navsip.dll" -Recurse
-        Write-Host "Found navsip at $($navsip.FullName)"
-        Copy-Item -Path $navsip.FullName -Destination $NavSipDestination -Force | Out-Null
-        Write-Host "Copied navsip to $NavSipDestination"
-    }
-    finally {
-        Remove-Item -Path $artifactTempFolder -Recurse -Force
-    }
+        if (Get-Command -Name "sign" -ErrorAction SilentlyContinue) {
```

Comment:
Doesn't this return true if there is a function called sign in the scope as well? Is this really the right way to check whether dotnet sign tool is installed?

Reply:
Wouldn't it be safer to just always install the signing tool we need locally and ignore any globally installed version and then set the alias - this way we know what we run.
Installing it locally just says it is already installed if it is:
![image](https://github.com/microsoft/AL-Go/assets/10775043/fe44db57-6813-44c2-8051-2dea17cb9660)